baking/src/tezos_baking/tezos_setup_wizard.py

```diff
@@ -387,7 +393,9 @@ def get_snapshot_link(self):
                             self.config["history_mode"] == "archive"
                             and artifact["history_mode"] == "full"
                         )
-                    ),
+                    )
+                    and artifact["tezos_version"]["commit_info"]["commit_hash"]
+                    == node_version_hash,
```

Comment:
I suppose that it's unlikely, but perhaps we should instead check that one hash is prefix of the other (or the other way around), in case only one of them uses the full hash.